src/screens/DomainScreen/index.js

```diff
@@ -101,20 +101,21 @@ const DomainScreen = () => {
   };
 
   const getIFollow = async () => {
-    if (ifollow.length === 0) {
+    const isFollow = JSON.stringify(ifollow.map((i) => i.domain_id_followed)).includes(iddomain);
+    if (!isFollow) {
       const res = await getDomainIdIFollow();
       setIFollow(res.data, dispatch);
     } else {
-      setFollow(JSON.stringify(ifollow).includes(iddomain));
+      setFollow(isFollow);
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for checking if a domain is followed has been changed. Previously, it was checked if the `ifollow` array is empty, and if so, a request was made to get the domains that the user follows. Now, it checks if the current domain is in the `ifollow` list, and if not, it makes a request to get the domains that the user follows. This change could potentially lead to unnecessary requests if the `ifollow` list is not empty but does not contain the current domain. Consider reverting this change or provide a more efficient way to check if the current domain is in the `ifollow` list.
```diff
-    const isFollow = JSON.stringify(ifollow.map((i) => i.domain_id_followed)).includes(iddomain);
-    if (!isFollow) {
+    if (ifollow.length === 0 || !ifollow.some(i => i.domain_id_followed === iddomain)) {
       const res = await getDomainIdIFollow();
       setIFollow(res.data, dispatch);
     } else {
       setFollow(isFollow);
     }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->